src/main/kotlin/org/rust/ide/intentions/FillMatchArmsIntention.kt

```diff
@@ -35,7 +35,16 @@ class FillMatchArmsIntention : RsElementBaseIntentionAction<FillMatchArmsIntenti
     override fun invoke(project: Project, editor: Editor, ctx: Context) {
         val (expr, name, variants) = ctx
         var body = RsPsiFactory(project).createMatchBody(name, variants)
-        body = (expr.matchBody?.replace(body) ?: expr.addAfter(body, expr.expr)) as RsMatchBody
+        val matchBody = expr.matchBody
+        if (matchBody != null) {
+            val rbrace = matchBody.rbrace
+            for (arm in body.matchArmList){
+                matchBody.addBefore(arm, rbrace)
```

Comment:
Looks like like we could use `matchBody.addRangeBefore(body.firstChild, body.lastChild, rbrace)` here (but then needs to check the corner case where enum body is empty)